htdocs/core/modules/facture/mod_facture_mars.php

```diff
@@ -179,7 +180,8 @@ function getNextValue($objsoc,$facture,$mode='next')
             $sql = "SELECT facnumber as ref";
             $sql.= " FROM ".MAIN_DB_PREFIX."facture";
             $sql.= " WHERE facnumber LIKE '".$prefix."____-".$num."'";
-            $sql.= " AND entity IN (".getEntity('invoicenumber').")";
+			if(!empty($conf->global->MULTICOMPANY_INVOICENUMBER_SHARING_ENABLED))  $sql.= " AND entity IN (".getEntity('invoicenumber').")";
+			else $sql.= " AND entity = $entity";
```

Comment:
@atm-quentin 
can you restore the original code please, i have another solution for this